### DISCUSSION TITLE
Simultaneous logging

### DISCUSSION THREAD

**Discussion by TornaxO7** [tokio-rs/tracing#2299] [Q&A] [answered]
Hello! I've got a question: Is it possible to log to multiple sources at the "same" time? For example: I'd like to log to a file but also to journald as well, is that possible with `tracing`?

**Comment by davidbarsky** [comment, marked as answer]
Yes, take a look at [this example](https://github.com/tokio-rs/tracing/blob/v0.1.x/examples/examples/fmt-multiple-writers.rs). To adopt it to your use-case, replace the [`fmt::Layer()` on line 18](https://github.com/tokio-rs/tracing/blob/v0.1.x/examples/examples/fmt-multiple-writers.rs#L18) with [`tracing_journald::Layer`](https://docs.rs/tracing-journald/latest/tracing_journald/index.html)::[Layer](https://docs.rs/tracing-journald/latest/tracing_journald/struct.Layer.html#). The code will look something like this:

```rust
use std::io;
use tracing_subscriber::{fmt, layer::SubscriberExt, EnvFilter};

fn main() {
    let file_appender = tracing_appender::rolling::hourly(dir, "example.log");
    let (non_blocking, _guard) = tracing_appender::non_blocking(file_appender);

    let subscriber = tracing_subscriber::registry()
        .with(EnvFilter::from_default_env().add_directive(tracing::Level::TRACE.into()))
        .with(tracing_journald::Layer::new())
        .with(fmt::Layer::new().with_writer(non_blocking));
    tracing::subscriber::set_global_default(subscriber).expect("Unable to set a global subscriber");
}
```

**Reply by TornaxO7** [reply]
Awesome! Thank you for the verification! c(^-^)c
